base_layer/core/src/chain_storage/tests/blockchain_database.rs

```diff
@@ -618,18 +619,13 @@ mod validator_node_merkle_root {
             .unwrap()
             .unwrap();
 
-        let mut vn_mmr = ValidatorNodeMmr::new(Vec::new());
-        vn_mmr
-            .push(
-                Blake256::new()
-                    .chain(public_key.as_bytes())
-                    .chain(shard_key.as_slice())
-                    .finalize()
-                    .to_vec(),
-            )
-            .unwrap();
+        let vn_bmt = ValidatorNodeBMT::create(vec![Blake256::new()
+            .chain(public_key.as_bytes())
+            .chain(shard_key.as_slice())
+            .finalize()
+            .to_vec()]);
```

Comment:
Unless there's a compelling reason not to, this should use a domain-separated hasher.

Reply:
This is just a test. The balanced tree is using domain-separated hasher. But as I'm looking at the code for the hash of VNs we are not using the domain-separated hasher there. But that's not part of this PR.

Reply:
Whoops, made this comment in the wrong spot. Sorry for the confusion.

Reply:
Opened an [issue](https://github.com/tari-project/tari/issues/5205) for this, so another PR can address it.